core/src/main/scala/org/bykn/bosatsu/LetFree.scala

```diff
@@ -336,15 +520,15 @@ object LetFreeConversion {
     val res = headReduction(expr) match {
       case App(fn, arg) =>
         App(normalOrderReduction(fn), normalOrderReduction(arg))
-      case extVar @ ExternalVar(_, _) => extVar
+      case extVar @ ExternalVar(_, _, _) => extVar
       // check for a match reduction opportunity (beta except for Match instead of lambda)
       case Match(arg, branches) =>
         Match(normalOrderReduction(arg), branches.map{ case (p, s) => (p, normalOrderReduction(s))})
       case lv @ LambdaVar(_)  => lv
       // check for eta reduction
       case Lambda(expr)       =>
         Lambda(normalOrderReduction(expr))
-      case Struct(enum, args) => Struct(enum, args.map(normalOrderReduction(_)))
+      case Struct(enum, args, df) => Struct(enum, args.map(normalOrderReduction(_)), df)
```

Comment:
could adding data family be a separate PR?

Reply:
It would just mean it was buggy between the PRs... but that seems fine.